context_test.go

```diff
@@ -41,6 +41,25 @@ func TestRespond(t *testing.T) {
 	}
 }
 
+func TestStatus(t *testing.T) {
+	c := NewContext()
+	data := map[string]string{"name": "jim"}
+	r := c.Status(201, data)
```

Comment:
Initially instead of creating a new method to send the response data with I treated this as a builder, ie:

```go
c.Status(201).Respond(data)
```

It works and might be a more complete method of setting the status code as more than just `Respond` actually send back data to the client. 

Looking for some input on if the current approach is good, or if it should be setup a different way.

Reply:
Yeah, I agree it is the interface that I prefer.

A user wouldn't be able to call `return c.Respond(data).Status(401)` (it would fail at the compiler, man I miss compilers) since `Respond()` return the Response not the Context.

I'll get this updated to simplify the interfaces to where we have `Context.Status(status int)`, `Context.Respond(obj interface{})`, `Context.Error(err error)`, and `Context.Fail(err error)` and remove the ones with the status codes.